### PR TITLE
fix: support rhm and azure cloud providers and check quota on marketp…

### PR DESCRIPTION
**Pull request by pb82**
Support `rhm` and `azure` cloud providers. Also check Quota for marketplace billing models.

Verification steps:

1. Additional unit tests should pass
2. Creation of rhm marketplace instances should succeed when using an organization that has a cloud account linked to it

ping @davidffrench

### CODE REVIEW

File: internal/kafka/internal/services/quota/ams_quota_service.go [hunk 1]
```diff
@@ -162,7 +179,7 @@ func (q amsQuotaService) getAvailableBillingModelFromKafkaInstanceType(externalI
 	return billingModel, nil
 }
 
-func (q amsQuotaService) getBillingModel(kafka *dbapi.KafkaRequest, instanceType types.KafkaInstanceType) (string, error) {
+func (q amsQuotaService) GetBillingModel(kafka *dbapi.KafkaRequest, instanceType types.KafkaInstanceType) (string, error) {
```
Comment (machi1990) [inline internal/kafka/internal/services/quota/ams_quota_service.go:182]: Is this change needed?

Comment (machi1990) [reply internal/kafka/internal/services/quota/ams_quota_service.go:182]: Thanks @pb82 since it is a unit test within the package it should be okay to still keep it a private and test it, what do you think?

**Comment by machi1990** [comment]
Left one minor comment but my knowledge in this are pretty basic so I'll let others review and approve the functionality.
Ping @miguelsorianod and @pawelpaszki as they worked on this recent.

File: internal/kafka/internal/services/quota/ams_quota_service.go (hunk 1 above)
Comment (pb82) [reply internal/kafka/internal/services/quota/ams_quota_service.go]: you're absolutely correct @machi1990 , i've updated this now.

File: internal/kafka/internal/services/quota/ams_quota_service_test.go [hunk 2]
```diff
@@ -48,6 +48,407 @@ var ocmClientMockWithCloudAccounts = &ocm.ClientMock{
...
+				},
+				kafkaConfig: &defaultKafkaConf,
+			},
+			wantErr:          false,
+			wantBillingModel: string(v1.BillingModelMarketplaceAWS),
+		},
+	}
+
+	RegisterTestingT(t)
+	for _, tt := range tests {
```
Comment (pawelpaszki) [inline internal/kafka/internal/services/quota/ams_quota_service_test.go:439]: I think the same pattern as here: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1089/files#diff-ace4f53881679319ea877b48a99b7466398f3187ef68ae0825bb1ba233632168R49 needs to be followed for the `tt` in the tests

**Comment by pawelpaszki** [comment]
one comment added. otherwise LGTM

File: internal/kafka/internal/services/quota/ams_quota_service_test.go (hunk 2 above)
Comment (pb82) [reply internal/kafka/internal/services/quota/ams_quota_service_test.go]: good catch, fixed
